packages/hardhat/helpers/events.ts

```diff
@@ -0,0 +1,20 @@
+import { Contract } from "@ethersproject/contracts";
+
+export const getEventArgument = async (
+  selectedFilter: string,
+  arg: number | string,
+  contract: Contract,
+  transactionHash: string
+): Promise<string> => {
+  return new Promise((resolve, reject) => {
```

Comment:
is it needed to return a `new promise`? since declaring it as an async function always return a promise

Reply:
In this case we need to create the Promise. As I understand the async logic on a function return handle a promise element without need to include an await but in this case if we don't include the Promise we don't even have something to resolve on the first place.